### PR TITLE
Connect ui

### PR DESCRIPTION
**Pull request by MelvinMallari**
For #90. UI detects MetaMask locking/unlocking using an interval. This is in accordance to [MetaMask document best practices](https://github.com/MetaMask/faq/blob/master/DEVELOPERS.md#ear-listening-for-selected-account-changes). Feel free to let me know if this is not preferred at the moment, I'll move on to less 'UI polish work' (Rob's sentiment) rather than revisit if it isn't.

### CODE REVIEW

File: src/components/ui/Navigation.vue [hunk 1]
```diff
@@ -52,5 +56,30 @@ export default class Navigation extends Vue {
     const web3Module = getModule(Web3Module, this.$store)
     enableEthereum(flashesModule, metaMaskModule, web3Module)
   }
+
+  get publicKey() {
+    return this.isAddressDefined ? ethereum.selectedAddress : ''
```
Comment (robrobbins) [inline src/components/ui/Navigation.vue]: I don't think `isAddressDefined()` needs to exist:

    get publicKey() {
        return ethereum.selectedAddress || ''
    }

how does ^ not work for how you are using it here?

File: src/components/ui/Navigation.vue [hunk 2]
```diff
@@ -52,5 +56,30 @@ export default class Navigation extends Vue {
...
+           (ethereum.selectedAddress !== '')
+  }
+
+  get isConnected() {
+    return this.accountUnlocked
+  }
+
+  private setUnlockCheck() {
+    const accountInterval = setInterval(async () => {
+      const isMetaMaskUnlocked = await ethereum._metamask.isUnlocked() // tslint:disable-line
```
Comment (robrobbins) [inline src/components/ui/Navigation.vue]: that looks weird AF. It's also failing on Travis...

File: src/components/ui/Navigation.vue [hunk 3]
```diff
@@ -52,5 +56,30 @@ export default class Navigation extends Vue {
...
+      this.accountUnlocked = isMetaMaskUnlocked ? true : false
+    }, 100)
+  }
+
+  private mounted() {
+    this.setUnlockCheck()
```
Comment (robrobbins) [inline src/components/ui/Navigation.vue]: so why are we doing this? I smell TRON

**Comment by thirtyaughtsix** [comment]
You guys can hash it out but I feel pretty strongly that the test is incomplete.  Code was added here and none of it is covered.  Shouldn't we test the conditional rendering and spy the click action on connect?

**Comment by thirtyaughtsix** [comment]
Force merging so we don't get held up.  @MelvinMallari has corrected all of @robrobbins requests.

**Pull request merged**